### PR TITLE
Refactor Game page and add animation on cards

### PR DESCRIPTION
**Pull request by dbertella**

### CODE REVIEW

File: packages/client/src/components/Cards/CardWrapper.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { PropsWithChildren, MouseEvent, useRef } from 'react';
+import { Box } from 'theme-ui';
+import { useSpring, animated, config } from 'react-spring';
+
+type CardWrapperProps = PropsWithChildren<{ onClick: (e: MouseEvent<HTMLElement>) => void; moving: boolean }>;
```
Comment (stathismor) [inline packages/client/src/components/Cards/CardWrapper.tsx]:
```suggestion
type CardWrapperProps = PropsWithChildren<{ onClick: (e: MouseEvent<HTMLElement>) => void; isMoving: boolean }>;
```

File: packages/client/src/components/Cards/CardWrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+const INITIAL_POSITION = { y: 0, x: 0 };
+
+export const CardWrapper = ({ moving, ...rest }: CardWrapperProps) => {
+  const ref = useRef<HTMLDivElement | null>(null);
+
+  const { x, y } = ref.current?.getBoundingClientRect() ?? INITIAL_POSITION;
```
Comment (stathismor) [inline packages/client/src/components/Cards/CardWrapper.tsx:12]: What do the `?` mean here...?

File: packages/client/src/components/Cards/CardWrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  const { xys } = useSpring({
+    config: config.slow,
+    xys: moving ? [window.innerWidth / 2 - x, window.innerHeight - y, 0.9] : [0, 0, 1],
+  });
+  console.log(moving, xys);
```
Comment (stathismor) [inline packages/client/src/components/Cards/CardWrapper.tsx]: :horse:

**Review by stathismor — APPROVED** [review]

File: packages/client/src/components/Cards/CardWrapper.tsx (hunk 2 above)
Comment (dbertella) [reply packages/client/src/components/Cards/CardWrapper.tsx:12]: This is optional chaining, it's a new feature in js/ts check it out https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

**Pull request merged**